docs/DeveloperGuide.md

```diff
@@ -330,14 +330,83 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `ContactCS` and the **Actor** is the `CS freshman`, unless specified otherwise)
+
+**Use case: Add a contact**
+
+**MSS**
+
+1.  User requests to add a new contact by inputting the relevant contact details (name, email, phone number, module code, etc.)
+2.  ContactCS verifies the input format and checks for duplicate
+3.  ContactCS adds the contact under the specified module
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. Necessary input such as module code, name, or email is missing.
+
+  * 2a1. ContactCS requests the user to provide the required information
+  * 2a2. User enters new data
+  * Steps 2a1 - 2a2 are repeated until the data entered are correct
+        
+    Use case resumes from step 2.
+
+* 3a. The given format is invalid.
+
+    * 3a1. ContactCS requests the user to provide the correct format and shows the valid command format
+    * 3a2. User enters new input
+    * Steps 3a1 - 3a2 are repeated until the format for the entered input is correct
+
+      Use case resumes from step 2.
+
+* 4a. The given contact is a duplicate.
```

Comment:
I think the step numbers are wrong for add contact use case extensions, the checks for valid format & emails are in step 2 hence extensions based on this step should be 2.*(For example, 2a, 2b, 2c etc.) You may refer here for correction: 
![image](https://github.com/user-attachments/assets/011799ef-a377-4e88-ab1d-107cb4a3a5de)

Edited: Also I realize, it looks weird and doesn't make sense to extend from step 3 and resumes from step 2, I guess you may just be confused about the numbering, so should definitely take note for rest of the parts as well, thanks